src/App.tsx

```diff
@@ -1,61 +1,112 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './react-app-env';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getTodoId = (todos: Todo[]) => {
+  const id = Math.max(...todos.map(todo => todo.id));
+
+  return id + 1;
+};
+
+const getUser = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+const todosAndUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosAndUsers);
+  const [title, setTitle] = useState('');
+  const [todoUserId, setTodoUserId] = useState(0);
+  const [hasError, setHasError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setTodoUserId(0);
+    setHasError(false);
+  };
+
+  const onFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoUserId || !title) {
+      setHasError(true);
+    } else {
```

Comment:
You don't need else branch. If you have an error - set errors and return immediately from function.